src/igvfd/types/file_set.py

```diff
@@ -526,40 +527,72 @@ def sample_summary(self, request, samples=None):
+            'pooled cell specimen, reprogrammed cell specimen': ['pooled reprogrammed cell specimen'],
+        }
+
+        classification_to_prefix = {
+            'differentiated cell specimen': 'differentiated',
+            'reprogrammed cell specimen': 'reprogrammed',
+            'pooled differentiated cell specimen': 'pooled differentiated',
+            'pooled reprogrammed cell specimen': 'pooled reprogrammed'
+        }
+
         for sample in samples:
             sample_object = request.embed(sample, '@@object')
 
             # Group sample and targeted sample terms according to classification.
             # Other metadata such as treatment info are lumped together.
-            classification = ' and '.join(sample_object['classifications'])
+            mux_prefix = ''
+            sample_classifications = sorted(sample_object['classifications'])
+            if 'multiplexed sample' in sample_object['classifications']:
+                sample_classifications.remove('multiplexed sample')
+                mux_prefix = 'multiplexed sample of '
+            if ', '.join(sorted(sample_classifications)) in two_classification_cases:
+                sample_classifications = two_classification_cases[', '.join(sorted(sample_classifications))]
+            # The variable "classification" can potentially be very long for
+            # a Multiplexed Sample, but it will be entirely dropped for
+            # Multiplexed Sample in the end - so it is ok.
+            classification = f"{mux_prefix}{' and '.join(sample_classifications)}"
             if classification not in sample_classification_term_target:
                 sample_classification_term_target[classification] = set()
 
             for term in sample_object['sample_terms']:
                 sample_term_object = request.embed(term, '@@object?skip_calculated=true')
+                sample_phrase = f"{sample_term_object['term_name']}"
+                # Avoid redundancy of classification and term name
+                # e.g. "HFF-1 cell cell line"
+                if not classification.startswith('multiplexed sample of'):
+                    if sample_phrase.endswith('cell') and 'cell' in classification:
+                        sample_phrase = sample_phrase.replace('cell', classification)
+                    elif 'cell' in sample_phrase and classification in classification_to_prefix:
+                        sample_phrase = f'{classification_to_prefix[classification]} {sample_phrase}'
+                    # if ' cell' in sample_phrase and 'cell' in classification:
+                    #     sample_phrase = sample_phrase.replace(' cell ', classification)
+                    # elif ' tissue' in sample_phrase and 'tissue' in classification:
+                    #     sample_phrase = sample_phrase.replace(' tissue ', classification)
+                    # elif ' gastruloid' in sample_phrase:
```

Comment:
would you lose the grastruloid specific replacements here? asking because gastruloid targeted_sample_term is always /sample-terms/UBERON_00004734/.  With your current code, it'll be "sample_term gastruloid induced to gastrula" instead of "sample_term induced to gastrula" (remove gastruloid).